core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java

```diff
@@ -815,6 +821,38 @@ public String toString() {
     });
   }
 
+  @Override
+  public Iterator<UfsStatus> listStatusIterable(
+      String path, ListOptions options, String startAfter,
+      int batchSize) throws IOException {
+    return call(new UfsCallable<Iterator<UfsStatus>>() {
+      @Override
+      public Iterator<UfsStatus> call() throws IOException {
+        Iterator<UfsStatus> result =
+            mUnderFileSystem.listStatusIterable(path, options, startAfter, batchSize);
+        return filterInvalidPaths(result, path);
+      }
+
+      @Override
+      public String methodName() {
+        return "ListStatusIterable";
+      }
+
+      @Override
+      public String toString() {
+        return String.format("path=%s, options=%s", path, options);
+      }
+    });
+  }
+
+  @Nullable
+  Iterator<UfsStatus> filterInvalidPaths(Iterator<UfsStatus> statuses, String listedPath) {
```

Comment:
this is probably the wrong place for this, since this class is only a wrapper to add logging

Reply:
yeah i agree but the current listStatus() method also calls filterInvalidPaths() in this call :(